### PR TITLE
Feature/12599 fss tooling add support for function operators in search

### PR DESCRIPTION
**Pull request by rvijayavel**
This PR contains below implementations
- Function operators support
- Exists/No Exists support
- Copyright year to dynamic
- Set default operator on attribute change

### CODE REVIEW

File: file-share-service-desktop-client/Modules/Search/SearchCriteriaViewModel.cs [hunk 1]
```diff
@@ -143,5 +160,46 @@ private void SetVisiblePropertyForAndControl()
...
+                foreach (var item in SearchCriteria)
+                {
+                    item.IsValueEnabled = item.Operator == null ||
+                                            (item.Operator != Operators.Exists &&
+                                            item.Operator != Operators.NotExists);
+
+                    if (!item.IsValueEnabled)
+                    {
+                        item.Value = string.Empty;
+                        break;
```
[the remark this answers is not in the document]
Comment (rvijayavel) [reply file-share-service-desktop-client/Modules/Search/SearchCriteriaViewModel.cs]: Removed as per suggestion.

File: file-share-service-desktop-client/Modules/Search/SearchCriteriaViewModel.cs [hunk 2]
```diff
@@ -24,6 +24,9 @@ public class SearchCriteriaViewModel : BindableBase, ISearchCriteriaViewModel
         private readonly Attribute[] systemAttributes;
         private IEnumerable<Attribute> availableAttributes;
 
+        private const string OPERATOR_PROPERTY = "Operator";
```
[the remark this answers is not in the document]
Comment (rvijayavel) [reply file-share-service-desktop-client/Modules/Search/SearchCriteriaViewModel.cs]: Removed the hardcode value and used name as suggested, in event. Thanks.

File: file-share-service-desktop-client/ViewModels/MainWindowViewModel.cs [hunk 3]
```diff
@@ -68,5 +69,7 @@ public void OnPageButtonExecute(IPageButton pageButton)
 
         public string Version => Assembly.GetExecutingAssembly().GetCustomAttributes<AssemblyFileVersionAttribute>()
             .Single().Version;
+
+        public int Year => DateTime.UtcNow.Year;
```
[the remark this answers is not in the document]
Comment (rvijayavel) [reply file-share-service-desktop-client/ViewModels/MainWindowViewModel.cs]: Done

File: file-share-service-desktop-client/Modules/Search/SearchCriteriaViewModel.cs [hunk 4]
```diff
@@ -24,6 +24,9 @@ public class SearchCriteriaViewModel : BindableBase, ISearchCriteriaViewModel
...
+        private const string SELECTED_FIELD_PROPERTY = "SelectedField";
```
[the remark this answers is not in the document]
Comment (rvijayavel) [reply file-share-service-desktop-client/Modules/Search/SearchCriteriaViewModel.cs]: Done

**Review by crdandoukho — APPROVED** [review]
Looks good. Thanks

**Review by rockydevnet — APPROVED** [review]
LGTM

**Pull request merged**